expression/schema.go

```diff
@@ -61,40 +64,37 @@ func (s Schema) FindColumn(astCol *ast.ColumnName) (*Column, error) {
 	if idx == -1 {
 		return nil, nil
 	}
-	return s[idx], nil
+	return s.Columns[idx], nil
 }
 
-// InitIndices sets indices for columns in schema.
-func (s Schema) InitIndices() {
-	for i, c := range s {
+// InitColumnIndices sets indices for columns in schema.
+func (s Schema) InitColumnIndices() {
+	for i, c := range s.Columns {
 		c.Index = i
 	}
 }
 
 // RetrieveColumn retrieves column in expression from the columns in schema.
 func (s Schema) RetrieveColumn(col *Column) *Column {
-	index := s.GetIndex(col)
+	index := s.GetColumnIndex(col)
 	if index != -1 {
-		return s[index]
+		return s.Columns[index]
 	}
 	return nil
 }
 
-// GetIndex finds the index for a column.
-func (s Schema) GetIndex(col *Column) int {
-	for i, c := range s {
+// GetColumnIndex finds the index for a column.
+func (s Schema) GetColumnIndex(col *Column) int {
+	for i, c := range s.Columns {
 		if c.FromID == col.FromID && c.Position == col.Position {
 			return i
 		}
 	}
 	return -1
 }
 
-//Schema2Exprs converts []*Column to []Expression.
-func Schema2Exprs(schema Schema) []Expression {
-	result := make([]Expression, 0, len(schema))
-	for _, col := range schema {
-		result = append(result, col.Clone())
-	}
-	return result
+// MergeSchema will merge two schema into one schema.
+func MergeSchema(lSchema, rSchema Schema) Schema {
```

Comment:
Merge Schema should change the argument itself. It's better to create a new schema